tests/unit/test_utils.py

```diff
@@ -386,6 +387,65 @@ def test_unpack_zip(self, data):
         self.confirm_files()
 
 
+def test_rmtree_errorhandler_not_existing_directory(tmpdir):
+    """
+    Test rmtree_errorhandler ignores the given non-existing directory.
+    """
+    not_existing_path = str(tmpdir / 'foo')
+    mock_func = Mock()
+    rmtree_errorhandler(mock_func, not_existing_path, None)
+    mock_func.assert_not_called()
+
+
+def test_rmtree_errorhandler_readonly_directory(tmpdir):
+    """
+    Test rmtree_errorhandler makes the given read-only directory writable.
+    """
+    # Create read only directory
+    path = str((tmpdir / 'subdir').mkdir())
+    os.chmod(path, stat.S_IREAD)
+
+    # Make sure mock_func is called with the given path
+    mock_func = Mock()
+    rmtree_errorhandler(mock_func, path, None)
+    mock_func.assert_called_with(path)
+
+    # Make sure the path is become writable
+    assert os.stat(path).st_mode & stat.S_IWRITE
+
+
+def test_rmtree_errorhandler_reraises_error(tmpdir):
+    """
+    Test rmtree_errorhandler reraises an exception
+    by the given non-readonly directory.
+    """
+    # Create directory without read permission
+    path = str((tmpdir / 'subdir').mkdir())
+    os.chmod(path, ~stat.S_IREAD)
+
+    mock_func = Mock()
+    try:
+        # Make sure the handler reraises an exception.
+        # Note that the raise statement without expression and
+        # active exception in the current scope throws
+        # the RuntimeError on Python3 and the TypeError on Python2.
```

Comment:
To avoid needing explanation and catching unintended errors, can we execute `rmtree_errorhandler` in an `except`? We can throw a custom exception or a `RuntimeError` with a specific message and then check the message to make sure it was the one we threw. Like

```
try:
    raise RuntimeError('test message')
except RuntimeError:
    with pytest.raises(...) as e:
        rmtree_errorhandler(...)
    assert e.value ...
```

Reply:
Good approach! I'll try it.

Reply:
Thanks! It works.